js/src/absolute-path-test.js

```diff
@@ -0,0 +1,98 @@
+// @flow
```

Comment:
Again, this and absolute-path.js are literal translations of the Go implementation.